src/main.rs

```diff
@@ -272,55 +279,67 @@ const APP: () = {
     /// Start a measurement for both the SHTCx sensor and the DS18B20 sensor.
     #[task(resources = [delay, sht, one_wire, ds18b20], schedule = [read_measurement_results])]
     fn start_measurements(mut ctx: start_measurements::Context) {
+        let mut measurement_plan = MeasurementPlan {
+            measure_shtc3: true,
+            measure_ds18b20: true,
+        };
         ctx.resources
             .sht
             .start_measurement(PowerMode::NormalMode)
-            .expect("SHTCx: Failed to start measurement");
+            .unwrap_or_else(|_| measurement_plan.measure_shtc3 = false);
         ctx.resources
             .ds18b20
             .start_measurement(&mut ctx.resources.one_wire, ctx.resources.delay)
-            .expect("DS18B20: Failed to start measurement");
+            .unwrap_or_else(|_| measurement_plan.measure_ds18b20 = false);
 
         // Schedule reading of the measurement results
         ctx.schedule
-            .read_measurement_results(Instant::now() + 500.millis())
+            .read_measurement_results(Instant::now() + 500.millis(), measurement_plan)
             .unwrap();
     }
 
     /// Read measurement results from the sensors. Re-schedule a measurement.
     #[task(resources = [debug, delay, sht, one_wire, ds18b20, rn], schedule = [start_measurements])]
-    fn read_measurement_results(mut ctx: read_measurement_results::Context) {
+    fn read_measurement_results(
+        mut ctx: read_measurement_results::Context,
+        measurement_plan: MeasurementPlan,
+    ) {
         static mut COUNTER: usize = 0;
 
         // Fetch measurement results
-        let shtc3_measurement = ctx
-            .resources
-            .sht
-            .get_raw_measurement_result()
-            .expect("SHTCx: Failed to read measurement");
-        let ds18b20_measurement = ctx
-            .resources
-            .ds18b20
-            .read_raw_temperature_data(&mut ctx.resources.one_wire, ctx.resources.delay)
-            .expect("DS18B20: Failed to read measurement");
+        let shtc3_measurement = if measurement_plan.measure_shtc3 {
+            ctx.resources.sht.get_raw_measurement_result().ok()
+        } else {
+            None
+        };
```

Comment:
This is a case where I think a match is easier to read:

```suggestion
        let shtc3_measurement = match measurement_plan.measure_shtc3 {
            true => ctx.resources.sht.get_raw_measurement_result().ok(),
            false => None,
        };
```

Not sure about the other `if` below, probably depends on what rustfmt does.

Reply:
Or we could even use an `Option<()>` and then just `and_then`?

Reply:
I think I'll just use https://doc.rust-lang.org/stable/std/primitive.bool.html#method.then :slightly_smiling_face: will bump our MRRV to 1.50.0 but I guess this is OK.

Reply:
Yep, sounds like a good plan!

Reply:
Hmm. I'd actually need https://doc.rust-lang.org/stable/std/primitive.bool.html#method.then_some, which is not yet stable. With `then` it looks a bit awkward:
```Rust
        let sht_measurement = measurement_plan.measure_sht.then(|| {}).and_then(|_| {
            ctx.resources.sht.get_raw_measurement_result().ok()

        });
```

Reply:
Or
```Rust
        let sht_measurement = measurement_plan.measure_sht.then(|| {
            ctx.resources.sht.get_raw_measurement_result().ok()

        }).flatten();
```

Reply:
I'll stay with the if :laughing:

Reply:
Yeah, makes sense!